upath/tests/conftest.py

```diff
@@ -301,42 +299,49 @@ def http_fixture(local_testdir, http_server):
 
 
 @pytest.fixture(scope="session")
-def webdav_server():
-    from wsgidav.wsgidav_app import WsgiDAVApp
-    from cheroot import wsgi
-
-    with tempfile.TemporaryDirectory() as tempdir:
-        host = "127.0.0.1"
-        port = 8090
-        app = WsgiDAVApp(
-            {
-                "host": host,
-                "port": port,
-                "provider_mapping": {"/": tempdir},
-                "simple_dc": {
-                    "user_mapping": {"*": {"USER": {"password": "PASSWORD"}}}
-                },
-            }
-        )
-        srvr = wsgi.Server(bind_addr=(host, port), wsgi_app=app)
-        srvr.prepare()
-        thread = threading.Thread(target=srvr.serve)
-        thread.daemon = True
-        thread.start()
+def webdav_server(tmp_path_factory):
+    try:
```

Comment:
What is the reaspn for making the imports and therefore the WebDAV tests optional?

Reply:
Common behaviour across the test suite:

- s3 tests are skipped if s3fs or moto is not installed 
https://github.com/fsspec/universal_pathlib/blob/0bdb3ca2e99e0a13dcc7429d0072a60909964cf7/upath/tests/conftest.py#L188 https://github.com/fsspec/universal_pathlib/blob/0bdb3ca2e99e0a13dcc7429d0072a60909964cf7/upath/tests/conftest.py#L155
- hdfs tests are skipped if pyarrow or htcluster is not installed https://github.com/fsspec/universal_pathlib/blob/0bdb3ca2e99e0a13dcc7429d0072a60909964cf7/upath/tests/conftest.py#L121 https://github.com/fsspec/universal_pathlib/blob/0bdb3ca2e99e0a13dcc7429d0072a60909964cf7/upath/tests/conftest.py#L98-L100
- gcp tests are skipped if gcsfs or docker is not installed https://github.com/fsspec/universal_pathlib/blob/0bdb3ca2e99e0a13dcc7429d0072a60909964cf7/upath/tests/conftest.py#L249 https://github.com/fsspec/universal_pathlib/blob/0bdb3ca2e99e0a13dcc7429d0072a60909964cf7/upath/tests/conftest.py#L221-L222
- http tests are skipped if fsspec's http dependencies are not installed https://github.com/fsspec/universal_pathlib/blob/0bdb3ca2e99e0a13dcc7429d0072a60909964cf7/upath/tests/implementations/test_http.py#L11-L13

This PR does the same for webdav and adlfs.

The smoke test session in the noxfile installs all those dependencies though. So in the CI and running via nox those tests are not optional.